aspnet/web-api/overview/advanced/calling-a-web-api-from-a-net-client.md

```diff
@@ -29,106 +29,141 @@ In this tutorial, a client app is written that consumes the following web API.
+
+`Install-Package Microsoft.AspNet.WebApi.Client`
+
+The preceding command adds the following NuGet packages to the project:
+
+* Microsoft.AspNet.WebApi.Client
+* Newtonsoft.Json
 
-[!code-console[Main](calling-a-web-api-from-a-net-client/samples/sample2.cmd)]
+Json.NET is a popular high-performance JSON framework for .NET.
 
 <a id="AddModelClass"></a>
 ## Add a Model Class
 
-Add the following class to the application:
+Examine the Product class:
 
-[!code-csharp[Main](calling-a-web-api-from-a-net-client/samples/sample3.cs)]
+[!code-csharp[Main](calling-a-web-api-from-a-net-client/sample/client/Program.cs?name=snippet_prod)]
 
-This class matches the data model used by the web API. We can use **HttpClient** to read a `Product` instance from an HTTP response, without having to write a lot of deserialization code.
+This class matches the data model used by the web API. An app can use **HttpClient** to read a `Product` instance from an HTTP response. The app doesn't have to write any deserialization code.
 
 <a id="InitClient"></a>
 ## Create and Initialize HttpClient
 
-Add a static **HttpClient** property to the `Program` class. 
+Examine the static **HttpClient** property:
 
-[!code-csharp[Main](calling-a-web-api-from-a-net-client/samples/sample4.cs)]
+[!code-csharp[Main](calling-a-web-api-from-a-net-client/sample/client/Program.cs?name=snippet_HttpClient)]
 
-> [!NOTE]
-> **HttpClient** is intended to be instantiated once and re-used throughout the life of an application. Especially in server applications, creating a new **HttpClient** instance for every request will exhaust the number of sockets available under heavy loads. This will result in **SocketException** errors.
+**HttpClient** is intended to be instantiated once and reused throughout the life of an application. The following conditions can result in **SocketException** errors:
 
+* Creating a new **HttpClient** instance per request.
+* Server under heavy load.
 
-To initialize the **HttpClient** instance, add the following code to the `RunAsync` method: 
+Creating a new **HttpClient** instance per request can exhaust the available sockets.
 
-[!code-csharp[Main](calling-a-web-api-from-a-net-client/samples/sample5.cs)]
+The following code initializes the **HttpClient** instance:
 
-This code sets the base URI for HTTP requests, and sets the Accept header to "application/json", which tells the server to send data in JSON format.
+[!code-csharp[Main](calling-a-web-api-from-a-net-client/sample/client/Program.cs?name=snippet5)]
+
+The preceding code:
+
+* Sets the base URI for HTTP requests. Change the port number to the port used in the server app. The app won't work unless port for the server app is used.
+* Sets the Accept header to "application/json". Setting this header tells the server to send data in JSON format.
 
 <a id="GettingResource"></a>
 ## Sending a GET request to retrieve a resource
```

Comment:
Sending a GET request to retrieve a resource
Send a GET request to retrieve a resource (no gerunds per mvc specs)